server/models/Collective.js

```diff
@@ -1621,8 +1621,17 @@ export default function(Sequelize, DataTypes) {
           return models.Member.update({ deletedAt: new Date() }, { where: { id: { [Op.in]: diffMemberIds } } });
         }
       })
-      .then(() => {
-        return Promise.map(members, member => {
+      .then(async () => {
+        const oldAdminMembers = await this.getMembers({
+          where: { role: { [Op.in]: [roles.ADMIN] } },
+        });
+        const adminMembers = members.filter(m => m.role === roles.ADMIN);
+        const diff = differenceBy(oldAdminMembers, adminMembers, 'role');
```

Comment:
Shouldn't we make the difference by `id` rather than `role`? All `role`s in `oldAdminMembers` and `adminMembers` are `ADMIN` right?